`alr update` improvements

mosteo opened the pull request:
* Add back the `--online` option to update index and solution in a single step
* New ability to say in the command line the crates to try to update: `alr update [crate1]...` When none given, all crates in current solution are valid update candidates.
   * Refactored update logic into `Alire` while implementing this change.

Up to now, updating a workspace dependencies was an all-or-nothing affair.

Fixes #237, fixes #251

mosteo committed:
Reimplement `alr update --online`

mosteo committed:
On demand update of selected dependencies

mosteo committed:
Test: on demand update of selected crate

Fabien-Chouteau commented on src/alr/alr-commands-update.adb:
```diff
@@ -1,42 +1,49 @@
+
+      for Crate of Allowed loop
+         if not Old.Releases.Contains (Crate) then
+            Reportaise_Wrong_Arguments ("Requested crate is not in solution: "
```
I feel like it would be more meaningful for the user to use `Requested crate is not a dependency` here. `Solution` is a bit internal vocabulary.

mosteo replied:
D'accord

Fabien-Chouteau commented on src/alr/alr-commands-update.adb:
```diff
@@ -113,9 +147,23 @@ package body Alr.Commands.Update is
-               & " regenerates the aggregate project building file found in"
-               & " <crate>" & GNAT.OS_Lib.Directory_Separator
-               & Alire.Paths.Working_Folder_Inside_Root));
+      .Append ("Resolves unpinned dependencies using available indexes"));
```
Is the "selective" per crate update explain in the help message?

mosteo replied:
It is now

mosteo committed:
Review fixes

Explicitly check that requested crate is not pinned, since it would never
result in an update.

mosteo commented:
Added a check that requested crates be not pinned, since no update would be found for them anyway.

mosteo committed:
Merge branch 'master' into feat/update
